libs/sdk-core/src/breez_services.rs

```diff
@@ -313,20 +317,63 @@ impl BreezServices {
         {
             Some(_) => Err(SendPaymentError::AlreadyPaid),
             None => {
-                self.persist_pending_payment(&parsed_invoice, amount_msat, req.label.clone())?;
-                let payment_res = self
-                    .node_api
-                    .send_payment(parsed_invoice.bolt11.clone(), req.amount_msat, req.label)
-                    .map_err(Into::into)
-                    .await;
-                let payment = self
-                    .on_payment_completed(
-                        parsed_invoice.payee_pubkey.clone(),
-                        Some(parsed_invoice),
-                        payment_res,
-                    )
-                    .await?;
-                Ok(SendPaymentResponse { payment })
+                let pending_payment =
+                    self.persist_pending_payment(&parsed_invoice, amount_msat, req.label.clone())?;
+                self.notify_event_listeners(BreezEvent::PaymentStarted {
+                    details: pending_payment.clone(),
+                })
+                .await?;
+
+                match req.pending_timeout_sec {
+                    Some(secs) => {
+                        let (tx, rx) = std::sync::mpsc::channel();
+                        let cloned = self.clone();
+                        tokio::spawn(async move {
+                            let payment_res = cloned
```

Comment:
Would it work to call `node_api.send_payment` just before this `match`, and then only use the `match` to control waiting for `on_payment_completed`?

That way we could get rid of some code duplication.

Reply:
I tried various ways to achieve this, in the end doing so adds complications sharing the future across threads. This seems the simplest solution to me